### PR TITLE
:sparkles: Adding localsnap test

### PR DESCRIPTION
**Pull request by lsrinivas-pure**
**What this PR does / why we need it**:
Creates a locasnapshot and deletes it using stork APIs

**Which issue(s) this PR fixes** (optional)
Closes #PTX-4927,PTX-4928

**Special notes for your reviewer**:

Localsnap logs:
INFO[2022-01-28 13:04:24] Successfully taken lock for trigger [localSnapShot]

STEP: Create and Validate LocalSnapshots
INFO[2022-01-28 13:04:24] Namespace : fio-localsnap-longevity-0-01-28-12h54m00s
STEP: create schedule policy for fio-localsnap app
INFO[2022-01-28 13:04:29] schedPolicy is localintervalpolicy already exists
INFO[2022-01-28 13:04:29] Waiting for 2 mins for Snapshots to be completed
STEP: get volumes for fio-localsnap app
INFO[2022-01-28 13:06:30] Got volume count : 6
INFO[2022-01-28 13:06:30] snapshotScheduleName : fio-data-fio-0-interval-schedule for volume: fio-data-fio-0
INFO[2022-01-28 13:06:31] snapStatuses count: 1
INFO[2022-01-28 13:06:31] Policy Type: Interval
INFO[2022-01-28 13:06:31] ScheduledVolumeSnapShot Name: fio-data-fio-0-interval-schedule-interval-2022-01-28-072759
INFO[2022-01-28 13:06:31] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 13:06:31] Got SnapData Name: k8s-volume-snapshot-d3abf31f-2acf-4622-beed-9d46d1394f53
INFO[2022-01-28 13:06:31] Snapshot Type: local
INFO[2022-01-28 13:06:31] Snapshot ID: 794757703566679172
INFO[2022-01-28 13:06:59] testAndSetEndpoint failed for 10.233.0.190: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.233.0.190:9020: i/o timeout"
INFO[2022-01-28 13:06:59] Getting new driver.
INFO[2022-01-28 13:07:02] Using 10.13.10.17:9020 as endpoint for portworx volume driver
DEBU[2022-01-28 13:07:03] Validating Stork snapshot
INFO[2022-01-28 13:07:03] snapshotScheduleName : fio-data-fio-1-interval-schedule for volume: fio-data-fio-1
INFO[2022-01-28 13:07:03] snapStatuses count: 1
INFO[2022-01-28 13:07:03] Policy Type: Interval
INFO[2022-01-28 13:07:03] ScheduledVolumeSnapShot Name: fio-data-fio-1-interval-schedule-interval-2022-01-28-072811
INFO[2022-01-28 13:07:03] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 13:07:04] Got SnapData Name: k8s-volume-snapshot-dda20c72-1f6f-4978-a1d0-c9ef2946001a
INFO[2022-01-28 13:07:04] Snapshot Type: local
INFO[2022-01-28 13:07:04] Snapshot ID: 53165195283886280
INFO[2022-01-28 13:07:32] testAndSetEndpoint failed for 10.233.0.190: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.233.0.190:9020: i/o timeout"
INFO[2022-01-28 13:07:32] Getting new driver.
INFO[2022-01-28 13:07:35] Using 10.13.10.17:9020 as endpoint for portworx volume driver
DEBU[2022-01-28 13:07:36] Validating Stork snapshot
INFO[2022-01-28 13:07:36] snapshotScheduleName : fio-data-fio-2-interval-schedule for volume: fio-data-fio-2
INFO[2022-01-28 13:07:36] snapStatuses count: 1
INFO[2022-01-28 13:07:36] Policy Type: Interval
INFO[2022-01-28 13:07:36] ScheduledVolumeSnapShot Name: fio-data-fio-2-interval-schedule-interval-2022-01-28-072828
INFO[2022-01-28 13:07:36] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 13:07:36] Got SnapData Name: k8s-volume-snapshot-015c94b0-df82-4625-ae6b-609d8f2a9c9f
INFO[2022-01-28 13:07:37] Snapshot Type: local
INFO[2022-01-28 13:07:37] Snapshot ID: 719833356076804370
INFO[2022-01-28 13:08:04] testAndSetEndpoint failed for 10.233.0.190: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.233.0.190:9020: i/o timeout"
INFO[2022-01-28 13:08:04] Getting new driver.
INFO[2022-01-28 13:08:08] Using 10.13.10.17:9020 as endpoint for portworx volume driver
DEBU[2022-01-28 13:08:09] Validating Stork snapshot
INFO[2022-01-28 13:08:09] snapshotScheduleName : fio-log-fio-0-interval-schedule for volume: fio-log-fio-0
INFO[2022-01-28 13:08:09] snapStatuses count: 1
INFO[2022-01-28 13:08:09] Policy Type: Interval
INFO[2022-01-28 13:08:09] ScheduledVolumeSnapShot Name: fio-log-fio-0-interval-schedule-interval-2022-01-28-072756
INFO[2022-01-28 13:08:09] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 13:08:10] Got SnapData Name: k8s-volume-snapshot-baa096c9-e139-4cb2-a401-6bb5135dcbe2
INFO[2022-01-28 13:08:10] Snapshot Type: local
INFO[2022-01-28 13:08:10] Snapshot ID: 105684064706502890
INFO[2022-01-28 13:08:37] testAndSetEndpoint failed for 10.233.0.190: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.233.0.190:9020: i/o timeout"
INFO[2022-01-28 13:08:37] Getting new driver.
INFO[2022-01-28 13:08:40] Using 10.13.10.17:9020 as endpoint for portworx volume driver
DEBU[2022-01-28 13:08:41] Validating Stork snapshot
INFO[2022-01-28 13:08:41] ScheduledVolumeSnapShot Name: fio-log-fio-0-interval-schedule-interval-2022-01-28-073756
INFO[2022-01-28 13:08:41] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 13:08:41] Got SnapData Name: k8s-volume-snapshot-e5e53356-c1ff-48a0-b774-bd18ee0a5eef
INFO[2022-01-28 13:08:42] Snapshot Type: local
INFO[2022-01-28 13:08:42] Snapshot ID: 592932027528603503
INFO[2022-01-28 13:09:09] testAndSetEndpoint failed for 10.233.0.190: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.233.0.190:9020: i/o timeout"
INFO[2022-01-28 13:09:09] Getting new driver.
INFO[2022-01-28 13:09:12] Using 10.13.10.17:9020 as endpoint for portworx volume driver
DEBU[2022-01-28 13:09:13] Validating Stork snapshot
INFO[2022-01-28 13:09:13] snapshotScheduleName : fio-log-fio-1-interval-schedule for volume: fio-log-fio-1
INFO[2022-01-28 13:09:14] snapStatuses count: 1
INFO[2022-01-28 13:09:14] Policy Type: Interval
INFO[2022-01-28 13:09:14] ScheduledVolumeSnapShot Name: fio-log-fio-1-interval-schedule-interval-2022-01-28-072805
INFO[2022-01-28 13:09:14] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 13:09:14] Got SnapData Name: k8s-volume-snapshot-1b661c0a-f5d7-4004-860a-ff47edb59669
INFO[2022-01-28 13:09:14] Snapshot Type: local
INFO[2022-01-28 13:09:14] Snapshot ID: 680568844719880115
INFO[2022-01-28 13:09:42] testAndSetEndpoint failed for 10.233.0.190: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.233.0.190:9020: i/o timeout"
INFO[2022-01-28 13:09:42] Getting new driver.
INFO[2022-01-28 13:09:45] Using 10.13.10.17:9020 as endpoint for portworx volume driver
DEBU[2022-01-28 13:09:46] Validating Stork snapshot
INFO[2022-01-28 13:09:46] ScheduledVolumeSnapShot Name: fio-log-fio-1-interval-schedule-interval-2022-01-28-073806
INFO[2022-01-28 13:09:46] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 13:09:47] Got SnapData Name: k8s-volume-snapshot-5a06aab9-e413-4553-aab6-0f8f8d071500
INFO[2022-01-28 13:09:47] Snapshot Type: local
INFO[2022-01-28 13:09:47] Snapshot ID: 572822008534654427
INFO[2022-01-28 13:10:15] testAndSetEndpoint failed for 10.233.0.190: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.233.0.190:9020: i/o timeout"
INFO[2022-01-28 13:10:15] Getting new driver.
INFO[2022-01-28 13:10:18] Using 10.13.10.17:9020 as endpoint for portworx volume driver
DEBU[2022-01-28 13:10:19] Validating Stork snapshot
INFO[2022-01-28 13:10:19] snapshotScheduleName : fio-log-fio-2-interval-schedule for volume: fio-log-fio-2
INFO[2022-01-28 13:10:19] snapStatuses count: 1
INFO[2022-01-28 13:10:19] Policy Type: Interval
INFO[2022-01-28 13:10:19] ScheduledVolumeSnapShot Name: fio-log-fio-2-interval-schedule-interval-2022-01-28-072832
INFO[2022-01-28 13:10:19] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 13:10:19] Got SnapData Name: k8s-volume-snapshot-3eb88f76-8bb6-4cc3-be85-fac5a21a28f8
INFO[2022-01-28 13:10:20] Snapshot Type: local
INFO[2022-01-28 13:10:20] Snapshot ID: 664812877590064280
INFO[2022-01-28 13:10:48] testAndSetEndpoint failed for 10.233.0.190: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.233.0.190:9020: i/o timeout"
INFO[2022-01-28 13:10:48] Getting new driver.
INFO[2022-01-28 13:11:01] Using 10.13.10.70:9020 as endpoint for portworx volume driver
DEBU[2022-01-28 13:11:02] Validating Stork snapshot
INFO[2022-01-28 13:11:02] ScheduledVolumeSnapShot Name: fio-log-fio-2-interval-schedule-interval-2022-01-28-073832
INFO[2022-01-28 13:11:02] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 13:11:02] Got SnapData Name: k8s-volume-snapshot-21bff5c1-1155-4982-a592-a1aa544b71c0
INFO[2022-01-28 13:11:03] Snapshot Type: local
INFO[2022-01-28 13:11:03] Snapshot ID: 1133046749792415743
INFO[2022-01-28 13:11:31] testAndSetEndpoint failed for 10.233.0.190: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.233.0.190:9020: i/o timeout"
INFO[2022-01-28 13:11:31] Getting new driver.
INFO[2022-01-28 13:11:35] Using 10.13.10.17:9020 as endpoint for portworx volume driver
DEBU[2022-01-28 13:11:36] Validating Stork snapshot
INFO[2022-01-28 13:11:36] Trigger Function completed for [localSnapShot]
INFO[2022-01-28 13:11:36] Successfully released lock for trigger [localSnapShot]

Delete localsnap logs:
INFO[2022-01-28 23:38:24] Successfully taken lock for trigger [deleteLocalSnapShot]

STEP: Delete Schedule Policy and LocalSnapshots and Validate
INFO[2022-01-28 23:38:24] Namespace : fio-localsnap-longevity-0-01-28-23h17m12s
STEP: delete schedule policy for fio-localsnap app
STEP: get volumes for fio-localsnap app
INFO[2022-01-28 23:38:25] snapshotScheduleName : fio-data-fio-0-interval-schedule for volume: fio-data-fio-0
INFO[2022-01-28 23:38:25] Policy Type: Interval
INFO[2022-01-28 23:38:25] ScheduledVolumeSnapShot Name: fio-data-fio-0-interval-schedule-interval-2022-01-28-175616
INFO[2022-01-28 23:38:25] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 23:38:26] [fio-localsnap] Destroyed Snapshot: fio-data-fio-0-interval-schedule-interval-2022-01-28-175616
INFO[2022-01-28 23:38:26] ScheduledVolumeSnapShot Name: fio-data-fio-0-interval-schedule-interval-2022-01-28-180616
INFO[2022-01-28 23:38:26] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 23:38:26] [fio-localsnap] Destroyed Snapshot: fio-data-fio-0-interval-schedule-interval-2022-01-28-180616
INFO[2022-01-28 23:38:26] snapshotScheduleName : fio-data-fio-1-interval-schedule for volume: fio-data-fio-1
INFO[2022-01-28 23:38:27] Policy Type: Interval
INFO[2022-01-28 23:38:27] ScheduledVolumeSnapShot Name: fio-data-fio-1-interval-schedule-interval-2022-01-28-175613
INFO[2022-01-28 23:38:27] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 23:38:27] [fio-localsnap] Destroyed Snapshot: fio-data-fio-1-interval-schedule-interval-2022-01-28-175613
INFO[2022-01-28 23:38:27] ScheduledVolumeSnapShot Name: fio-data-fio-1-interval-schedule-interval-2022-01-28-180614
INFO[2022-01-28 23:38:27] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 23:38:28] [fio-localsnap] Destroyed Snapshot: fio-data-fio-1-interval-schedule-interval-2022-01-28-180614
INFO[2022-01-28 23:38:28] snapshotScheduleName : fio-data-fio-2-interval-schedule for volume: fio-data-fio-2
INFO[2022-01-28 23:38:28] Policy Type: Interval
INFO[2022-01-28 23:38:28] ScheduledVolumeSnapShot Name: fio-data-fio-2-interval-schedule-interval-2022-01-28-175611
INFO[2022-01-28 23:38:28] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 23:38:29] [fio-localsnap] Destroyed Snapshot: fio-data-fio-2-interval-schedule-interval-2022-01-28-175611
INFO[2022-01-28 23:38:29] ScheduledVolumeSnapShot Name: fio-data-fio-2-interval-schedule-interval-2022-01-28-180614
INFO[2022-01-28 23:38:29] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 23:38:29] [fio-localsnap] Destroyed Snapshot: fio-data-fio-2-interval-schedule-interval-2022-01-28-180614
INFO[2022-01-28 23:38:29] snapshotScheduleName : fio-log-fio-0-interval-schedule for volume: fio-log-fio-0
INFO[2022-01-28 23:38:30] Policy Type: Interval
INFO[2022-01-28 23:38:30] ScheduledVolumeSnapShot Name: fio-log-fio-0-interval-schedule-interval-2022-01-28-175613
INFO[2022-01-28 23:38:30] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 23:38:30] [fio-localsnap] Destroyed Snapshot: fio-log-fio-0-interval-schedule-interval-2022-01-28-175613
INFO[2022-01-28 23:38:30] ScheduledVolumeSnapShot Name: fio-log-fio-0-interval-schedule-interval-2022-01-28-180614
INFO[2022-01-28 23:38:30] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 23:38:30] [fio-localsnap] Destroyed Snapshot: fio-log-fio-0-interval-schedule-interval-2022-01-28-180614
INFO[2022-01-28 23:38:31] snapshotScheduleName : fio-log-fio-1-interval-schedule for volume: fio-log-fio-1
INFO[2022-01-28 23:38:31] Policy Type: Interval
INFO[2022-01-28 23:38:31] ScheduledVolumeSnapShot Name: fio-log-fio-1-interval-schedule-interval-2022-01-28-175615
INFO[2022-01-28 23:38:31] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 23:38:31] [fio-localsnap] Destroyed Snapshot: fio-log-fio-1-interval-schedule-interval-2022-01-28-175615
INFO[2022-01-28 23:38:31] ScheduledVolumeSnapShot Name: fio-log-fio-1-interval-schedule-interval-2022-01-28-180616
INFO[2022-01-28 23:38:31] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 23:38:32] [fio-localsnap] Destroyed Snapshot: fio-log-fio-1-interval-schedule-interval-2022-01-28-180616
INFO[2022-01-28 23:38:32] snapshotScheduleName : fio-log-fio-2-interval-schedule for volume: fio-log-fio-2
INFO[2022-01-28 23:38:33] Policy Type: Interval
INFO[2022-01-28 23:38:33] ScheduledVolumeSnapShot Name: fio-log-fio-2-interval-schedule-interval-2022-01-28-175615
INFO[2022-01-28 23:38:33] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 23:38:33] [fio-localsnap] Destroyed Snapshot: fio-log-fio-2-interval-schedule-interval-2022-01-28-175615
INFO[2022-01-28 23:38:33] ScheduledVolumeSnapShot Name: fio-log-fio-2-interval-schedule-interval-2022-01-28-180615
INFO[2022-01-28 23:38:33] ScheduledVolumeSnapShot status: Ready
INFO[2022-01-28 23:38:33] [fio-localsnap] Destroyed Snapshot: fio-log-fio-2-interval-schedule-interval-2022-01-28-180615
INFO[2022-01-28 23:38:44] Trigger Function completed for [deleteLocalSnapShot]
INFO[2022-01-28 23:38:44] Successfully released lock for trigger [deleteLocalSnapShot]

### CODE REVIEW

**Comment by lsrinivas-pure** [comment]
tp-dev run: https://jenkins.portworx.dev/job/Torpedo/job/tp-dev-test/982/console

File: drivers/scheduler/k8s/k8s.go [hunk 1]
```diff
@@ -2604,6 +2604,38 @@ func (k *K8s) ValidateVolumes(ctx *scheduler.Context, timeout, retryInterval tim
 	return nil
 }
 
+// GetSnapShotData retruns the snapshotdata
+func (k *K8s) GetSnapShotData(ctx *scheduler.Context, snapshotName, snapshotNameSpace string) (*snapv1.VolumeSnapshotData, error) {
+
+	snap, err := k8sExternalStorage.GetSnapshot(snapshotName, snapshotNameSpace)
+	if err != nil {
+		return nil, &scheduler.ErrFailedToGetVolumeParameters{
```
Comment (Rohit-PX) [inline drivers/scheduler/k8s/k8s.go:2612]: Error should be of type `ErrFailedToGetSnapshot`

File: drivers/scheduler/k8s/k8s.go [hunk 2]
```diff
@@ -2604,6 +2604,38 @@ func (k *K8s) ValidateVolumes(ctx *scheduler.Context, timeout, retryInterval tim
...
+			App:   ctx.App,
+			Cause: fmt.Sprintf("failed to get Snapshot: %v. Err: %v", snapshotName, err),
+		}
+	}
+
+	snapDataName := snap.Spec.SnapshotDataName
+	logrus.Infof("Got SnapData Name: %v", snapDataName)
+	if len(snapDataName) == 0 {
+		return nil, &scheduler.ErrFailedToGetVolumeParameters{
```
Comment (Rohit-PX) [inline drivers/scheduler/k8s/k8s.go:2621]: Error should be of type  `ErrFailedToGetSnapshotDataName`

**Review by Rohit-PX — CHANGES_REQUESTED** [review]
LGTM apart from these.

File: drivers/scheduler/k8s/k8s.go (hunk 2 above)
Comment (lsrinivas-pure) [reply drivers/scheduler/k8s/k8s.go]: fixed

File: drivers/scheduler/k8s/k8s.go (hunk 1 above)
Comment (lsrinivas-pure) [reply drivers/scheduler/k8s/k8s.go]: fixed

File: drivers/scheduler/k8s/k8s.go [hunk 3]
```diff
@@ -2604,6 +2604,38 @@ func (k *K8s) ValidateVolumes(ctx *scheduler.Context, timeout, retryInterval tim
...
+			App: ctx.App,
+			Cause: fmt.Sprintf("snapshot: [%s] %s does not have snapshotdata set",
+				snap.Metadata.Namespace, snap.Metadata.Name),
+		}
+	}
+
+	snapData, err := k8sExternalStorage.GetSnapshotData(snapDataName)
+	if err != nil {
+		return nil, &scheduler.ErrFailedToGetVolumeParameters{
```
[the remark this answers is not in the document]
Comment (lsrinivas-pure) [reply drivers/scheduler/k8s/k8s.go]: fixed

**Review by lsrinivas-pure — COMMENTED** [review]
updated review comments

**Review by Rohit-PX — APPROVED** [review]
LGTM.

**Pull request merged**